src/compiler/generate-common-tests.xsl

```diff
@@ -633,6 +645,46 @@
       </xsl:apply-templates>
    </xsl:template>
 
+   <!-- Generate error message for user-defined usage of names in XSpec namespace.
+        Context node is an x:variable element. -->
+   <xsl:template name="x:detect-reserved-variable-name" as="empty-sequence()">
+      <xsl:variable name="msg" as="xs:string"
+         select="'User-defined XSpec variables must not use the XSpec namespace.'"/>
+      <xsl:choose>
+         <xsl:when test="starts-with(normalize-space(@name),'Q{')">
+            <!-- URI-qualified name -->
+            <xsl:if test="replace(@name,'(^\s*Q\{)|(\}.*$)','') eq $xspec-namespace">
+               <xsl:sequence select="error(xs:QName('x:XSPEC008'), $msg)"/>
+            </xsl:if>
+         </xsl:when>
+         <xsl:when test="namespace-uri-from-QName(resolve-QName(@name,.)) eq $xspec-namespace">
+            <xsl:sequence select="error(xs:QName('x:XSPEC008'), $msg)"/>
```

Comment:
We'd better have a test for this error. I added a simple one directly to this pull request branch: 9f0cf39e0d9ebcdeabd1d4f744c430743ebca742 ([Travis](https://travis-ci.org/xspec/xspec/jobs/576373056#L1761), [AppVeyor](https://ci.appveyor.com/project/xspec/xspec/builds/26939231/job/gl0saonl5gjwlpwa#L572))
Feel free to modify it in any way.

Reply:
Thank you for adding that! It looks good to me.